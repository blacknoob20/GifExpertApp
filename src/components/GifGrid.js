// import React, { Fragment, useEffect, useState } from 'react';
import React, { Fragment } from 'react';
import { GifGridItem } from './GifGridItem';
// import { getGifs } from './helpers/getGifs';
import { useFetchGifs } from './hooks/useFetchGifs';

export const GifGrid = ({categoria}) => {
    // const [imgs, setImgs] = useState([]);
    const {data:imgs,loading} = useFetchGifs(categoria);

    // useEffect(()=>{
    //     getGifs(categoria)
    //         .then(setImgs);
    // }, [categoria]);

    // getGifs();

        return(
            <Fragment>
                <h3>{categoria}</h3>
                {loading && <p>Loading</p>}
                <div className="card-grid">
                    { imgs.map((img) =>(<GifGridItem key={img.id} {...img}/>))}
                </div>
            </Fragment >
        );
}