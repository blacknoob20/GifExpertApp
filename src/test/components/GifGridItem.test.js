import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
    const title = 'Prueba Gif Item';
    const url   = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('Debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');

        console.log();
        expect(div.prop('className').includes('animate__bounce')).toBe(true);
        // expect(div.props().className).toEqual(expect.stringMatching(/^(animate__bounce)/));
    })

});
