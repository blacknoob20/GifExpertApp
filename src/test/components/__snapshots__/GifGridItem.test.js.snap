// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Prueba Gif Item"
    src="https://localhost/algo.jpg"
  />
  <p>
    Prueba Gif Item
  </p>
</div>
`;
